src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/RazorSemanticTokensEndpoint.cs

```diff
@@ -62,11 +62,6 @@ internal class RazorSemanticTokensEndpoint : ISemanticTokensRangeEndpoint
 
         public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
-            if (!_languageServerFeatureOptions.RegisterBuiltInFeatures)
```

Comment:
Side fix: Semantic tokens is surprisingly not one of the handlers O# recognizes by default, so we actually need to register the capability here